applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -545,10 +548,16 @@ class MapperVertexMorphingMatrixFree : public Mapper
     void InitializeMappingVariables()
     {
         const unsigned int origin_node_number = mrOriginModelPart.Nodes().size();
-        mValuesOrigin.resize(3,ZeroVector(origin_node_number));
+        mValuesOrigin.resize(3);
```

Comment:
Out of curiosity, why was this necessary?

Reply:
It is because, it resizes the `mValuesOrigin` in the first iteration to a `std::vector` of size 3 having all the elements with `Vector` with size `origin_node_number`. With remeshing, the `origin_node_number` changes. So in the second iteration this again resizes the `mValuesOrigin` to size 3, which is already of that size. So it does not reinitialize its elements with new `origin_node_number` value coming after remeshing. This does not raise segmentation fault also, and gives only garbage values when used.